### ISSUE TITLE
Comment Preview

### ISSUE THREAD

**Issue by getmechamot** [getmechamot/website-Balloon-Juice-public#26]
Add Comment Preview.

**Comment by getmechamot** [comment]
from a commenter: my kingdom for a “Preview” function. The 5 minute edit window isn’t really a substitute – especially when the post ends up in the dungeon. A Preview that flags verboten words would be nice too.

**Comment by getmechamot** [comment]
Plugins look too old; custom dev looks pretty easy:

https://premium.wpmudev.org/blog/jquery-comment-preview-plugin-wordpress/
http://wpapi.com/add-the-comment-preview-to-comment-box-without-wordpress-plugin/

**Comment by thearchduke** [comment]
One of the reasons that people were clamoring for this feature is because so many posts just vanish afterwards due to arbitrary rulings by the spam filter. Something like this would show the formatting, and any issues with closed tags and stuff, but wouldn't help people out with that.

A quick google did indeed bring up some plugins that do this the right way, by seeing exactly how the backend is going to mangle the comment with an AJAX call. But they're also, yes, hopelessly outdated.

Do we actually know the rules that the spam filter is operating under? It would be easy enough to reproduce them in a javascript-based preview.

**Comment by getmechamot** [comment]
Yes. I’ll send you a link to a Dropbox text file; let me know as soon as you get it downloaded so I can clear it out; no reason to leave any cookies out there for clever rats to find and make use of.

 

This is the kind of thing that should be in the repo, but like I said, what I have right now is a work in progress and I don’t want to open the doors until it’s in better shape!

 

So the general process works like this:

Comment comes in, gets screened by WordPress rules, then Akismet analyzes and returns it with a score and then WordPress puts the comment in view, Pending (in moderation), Spam, or Trash.

 

We cannot control Akismet, and it was throwing some bad results recently, putting things into Trash which is usually quite rare. No one else looks in the Trash but me or John, and I rescue what I can but I’m very late or miss lots/most because it’s not my job.

 

But if we can have comments pass the rules from WordPress, that would likely take care of almost all the issues that aren’t Akismet. And now since I have a relationship with them (we paid for a license to get support on the recent issue), they let me know that they can fine-tune the mechanism for us should we get more unexpected trashed comments.  


It was a revelation last year when I worked with a user to find out why all his comments, from any device, ended up in the trash. Turns out one of the filters was “no.gmail.com” and so anyone with a gmail address that ended in “no” was auto-trashed. 

 

So I’ve solved a few such issues since. I hypothesize that commenters using the name of a real person gets bumped by Akismet but that’s just  that, no proof. So I encourage folks with nyms or email addresses to try changing them a bit and see if things work. And we narrow things down from there.

 

So putting those words/things into a Preview/check would be a great service to all!

 

Dropbox link coming in just a few minutes.

 

 

 

 

 

 

Comments disappearing is almost always trash. And 

G. Alain Chamot

719-423-0140

please@getmechamot.com

**Comment by getmechamot** [comment]
https://www.dropbox.com/s/m5jc3wpw2zxgq4h/terms.txt?dl=0

 

 

G. Alain Chamot

719-423-0140

please@getmechamot.com

**Comment by thearchduke** [comment]
Got it.
